File: module/overlay-apk/java/com/hivirtus/zygiskmode/floatsvc/BootReceiver.java
package com.hivirtus.zygiskmode.floatsvc;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

public final class BootReceiver extends BroadcastReceiver {

    private static final String TAG = "VirtusFloat";

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.i(TAG, "BootReceiver — starting FloatService");
        Intent svc = new Intent(context, FloatService.class);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(svc);
        } else {
            context.startService(svc);
        }
    }
}

File: overlay-app/app/src/main/java/com/hivirtus/zygiskmode/BootReceiver.kt
package com.hivirtus.zygiskmode

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent?) {
        when (intent?.action) {
            Intent.ACTION_BOOT_COMPLETED,
            Intent.ACTION_LOCKED_BOOT_COMPLETED,
            ACTION_MODULE_BOOT -> OverlayBootstrap.start(context)
        }
    }

    companion object {
        const val ACTION_MODULE_BOOT = "com.hivirtus.zygiskmode.BOOT_START"
    }
}
